internal/credentials/credentials.go

```diff
@@ -81,21 +81,21 @@ type CredentialV0 struct {
 	ApiKey string `json:"apiKey"`
 }
 
-func (c *Credential) ConflictCheck(compareWith Credential) error {
-	if c.ServerType == server_type.ServerTypeConnectCloud {
+func (c *Credential) ConflictCheck(newCred Credential) error {
+	if newCred.ServerType == server_type.ServerTypeConnectCloud {
```

Comment:
One suggestion, not necessarily to be done in this PR but since server type now determines code flows and will be compared in many places, WDYT of extending `ServerType` with a few methods like `ServerType.isCloud()`, `ServerType.isConnect()`?

Maybe something similar to creds? the comment below with `// this is a Connect Cloud credential` suggests that it might be possible to abstract that to have a `.isCloudCredential()` too.